bin/install_npm_packages_clean

```diff
@@ -0,0 +1,71 @@
+#!/bin/bash -eu
+
+# Ensure package.json is present
+if [ ! -f package.json ]; then
+    echo "No package.json found!"
+    exit 1
+fi
+
+# Ensure package-lock.json is present
+# TODO: What to do with yarn-based projects? Make the hash on lockfile optional?
+if [ ! -f package-lock.json ]; then
+    echo "No package-lock.json found!"
+    exit 1
+fi
+
+PLATFORM=$(uname -s)
+ARCHITECTURE=$(uname -m)
+NODE_VERSION=$(node --version)
+PACKAGE_HASH=$(hash_file package-lock.json)
+
+# TODO: This comes from Studio. Figure out if it's standard in Node or not.
+if [ -d patches ]; then
+  PATCHES_HASH=$(hash_directory patches/)
+else
+  PATCHES_HASH=nopatch
+fi
```

Comment:
Even if it isn't a standard approach, it might be worth leaving it here as it adds basically no overhead

Reply:
FYI the implementation was inspired by this recommendation: https://www.npmjs.com/package/patch-package#circleci ; so I guess we could consider that "standard" in that regard?

Reply:
Oh, nice. 

I'll add a link to it and remove the todo. Thanks!